packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/utils.ts

```diff
@@ -0,0 +1,127 @@
+import { UniqueIdentifier } from "@dnd-kit/core";
+import { arrayMove } from "@dnd-kit/sortable";
+import { groupBy } from "lodash";
+import { AccountPage } from "../../../../components/hooks/useAccountPages";
+
+export interface PageList {
+  treeList: TreeElement[];
+  flatList: TreeElement[];
+}
+
+export interface TreeElement extends AccountPage {
+  parentId: string;
+  depth: number;
+  index: number;
+  expanded: boolean;
+  expandable: boolean;
+  collapsed: boolean;
+  children?: TreeElement[];
+}
+
+const getRecursivePageList = (
+  groupedPages: { [id: string]: AccountPage[] },
+  id: string,
+  expandedIds: string[],
+  depth = 0,
+  collapsed = false,
+): PageList => {
+  const emptyTreeList = [] as TreeElement[];
+  const emptyList: PageList = {
+    treeList: emptyTreeList,
+    flatList: emptyTreeList,
+  };
+
+  return (
+    groupedPages[id]?.reduce((prev, page, index) => {
+      const expanded = expandedIds.includes(page.entityId);
+      const children = getRecursivePageList(
+        groupedPages,
+        page.entityId,
+        expandedIds,
+        depth + 1,
+        collapsed || !expanded,
+      );
+      const expandable = !!children.treeList.length;
+
+      const item = {
+        ...page,
+        depth,
+        index,
+        parentId: page.parentPageEntityId,
+        expanded,
+        expandable,
```

Comment:
This is calculated state, is it really necessary to have it here?

Reply:
Removed in https://github.com/hashintel/hash/commit/5cab3ac1a6df55c19655ea81a335bce2e28caeac